ddd-kata.md

```diff
@@ -0,0 +1,197 @@
+* As a moderator, strife for consensus and alignment in the team
+* Plan your time carefully. Timebox your efforts.
+* Most importantly, have fun modelling.
+
+#### How to run an Event Storming Workshop
+
+Event storming is a collaborative workshop method that helps to explore complex business domains and model them using events. An event is an action that occurred in the system at a specific time. Event storming can be conducted in two phases: a high-level event storm to identify the domains and then a detailed event storm to focus on a core domain.
+
+The high-level event storm consists of these steps:
+
+1. **Chaotic Exploration**: Invite the right people from different perspectives and roles, such as domain experts, developers, testers, etc. Provide unlimited modeling space, such as a wall or a whiteboard. Ask everyone to write down relevant events in the domain and the process that they are analyzing on sticky notes. Use different colors for different types of events, such as past, present, future, etc. Encourage everyone to participate and share their ideas without filtering or criticizing.
+2. **Enforce the Timeline**: Arrange the sticky notes on the modeling space in a chronological order, from left to right. Use a horizontal line to separate different processes or subdomains. Identify any gaps, duplicates, inconsistencies or ambiguities in the events and resolve them with the help of the participants. Use arrows or lines to show the causal relationships between events.
+3. **Enforce Consistency**: Review the timeline and check if it makes sense and reflects the reality of the domain. Ask questions such as "What happens next?", "What triggers this event?", "Who is involved in this event?", etc. Use different symbols or colors to mark different aspects of the events, such as commands, aggregates, policies, external systems, users, etc. Add any missing or new events that emerge from the discussion.
+4. **Identify the Bounded Contexts**: Group the events into logical clusters that represent different subdomains or areas of responsibility in the domain. These are called bounded contexts. A bounded context is a coherent and consistent set of events that share a common language and meaning. Use different shapes or colors to highlight the boundaries of each bounded context. Name each bounded context and label its main purpose or goal.
+5. **Detailed Design**: Zoom in on the core domain and refine its events and relationships. Use different symbols or colors to mark different elements of the domain model, such as entities, value objects, aggregates, services, repositories, factories, etc. Define the properties and behaviors of each element and how they interact with each other. Use scenarios or examples to validate and test the model. Identify any constraints, rules, policies or assumptions that affect the model and document them. Use the detailed design also to validate your bounded contexts.
```

Comment:
> Use different symbols or colors to mark different elements of the domain model, such as entities, value objects, aggregates, services, repositories, factories, etc.

I'd challenge this, see previous comment. I'd rather let people stick to the official grammar before getting creative. Gotta know the rules to break them and so on...

Reply:
Agreed